starter/src/test/java/dev/openbanking4/spring/security/multiauth/configurers/AuthCollectorFilterTest.java

```diff
@@ -0,0 +1,440 @@
+                "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0b3RvIiwiZ3JvdXAiOlsiYWRtaW4iLCJjbHViRmFsYWZlbCJdfQ.kQDyIfKsf7xM0uN2F47X_yJgtgnG8CDYtPc6t5KAHL8")});
+
+        authCollectorFilter.doFilterInternal(mockedRequest, Mockito.mock(HttpServletResponse.class), Mockito.mock(FilterChain.class));
+
+        //Then
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        assertThat(authentication).isNotNull();
+
+        UserDetails userDetailsExpected = User.builder()
+                .username("toto")
+                .password("")
+                .authorities(Stream.of(
+                        new SimpleGrantedAuthority("admin"),
+                        new SimpleGrantedAuthority("clubFalafel"),
+                        new SimpleGrantedAuthority("bobPower")
+                ).collect(Collectors.toList()))
+                .build();
+        UserDetails userDetailsResult = (UserDetails) authentication.getPrincipal();
+
+        assertThat(userDetailsResult.getUsername()).isEqualTo(userDetailsExpected.getUsername());
+        assertThat(userDetailsResult.getAuthorities()).isEqualTo(userDetailsExpected.getAuthorities());
+    }
+
+    @Test
+    public void testCookieAndAccessToken() throws Exception {
+        //Given
+        SecurityContextHolder.getContext().setAuthentication(null);
+        List<AuthCollector> authCollectors = Stream.of(
+                customJwtCookieCollector,
+                psd2Collector,
+                statelessAccessTokenCollector,
+                staticUserCollector
+        ).collect(Collectors.toList());
+        AuthCollectorFilter authCollectorFilter = new AuthCollectorFilter(authCollectors, authCollectors);
+
+        //When
+        HttpServletRequest mockedRequest = Mockito.mock(HttpServletRequest.class);
+        RequestContextHolder.setRequestAttributes(new ServletWebRequest(mockedRequest));
+
+        when(mockedRequest.getCookies()).thenReturn(new Cookie[]{new Cookie("sso",
+                "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0b3RvIiwiZ3JvdXAiOlsiYWRtaW4iLCJjbHViRmFsYWZlbCJdfQ.kQDyIfKsf7xM0uN2F47X_yJgtgnG8CDYtPc6t5KAHL8")});
+        when(mockedRequest.getHeader("Authorization")).thenReturn(
+                "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzY29wZSI6WyJhY2NvdW50cyIsInBheW1lbnRzIl19.NpzyDWjuoC9oAE50fb3W2Mgs1ORWuWYCMv2xg677fGc");
+
+
+        authCollectorFilter.doFilterInternal(mockedRequest, Mockito.mock(HttpServletResponse.class), Mockito.mock(FilterChain.class));
+
+        //Then
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        assertThat(authentication).isNotNull();
+
+        UserDetails userDetailsExpected = User.builder()
+                .username("toto")
+                .password("")
+                .authorities(Stream.of(
+                        new SimpleGrantedAuthority("admin"),
+                        new SimpleGrantedAuthority("clubFalafel"),
```

Comment:
Could do some marketing for falafel king here 👀

Reply:
fixed